### ISSUE TITLE
Support styled-jss

### ISSUE THREAD

**Issue by hendrul** [jxnblk/styled-system#81]
[styled-jss](https://github.com/cssinjs/styled-jss]) is near to have support for [function rules](http://cssinjs.org/json-api?v=v9.6.0#function-rules) in this [PR](https://github.com/cssinjs/styled-jss/pull/35), the added docs [here](https://github.com/cssinjs/styled-jss/blob/2dd5de959c782ef23c01d1780632a3a336342e66/README.md#composable-styles) explain how to add multiple function rules, like the great ones you offer in this wonderful lib... is it posible to support their theming api?... In my case I'm using [material-ui v1](https://material-ui-next.com/) and they select react-jss as their CSSinJS engine of choice for performance reasons (and other reasons) they explain [here](https://github.com/mui-org/material-ui/issues/6115) and [here](https://github.com/mui-org/material-ui/blob/master/ROADMAP.md#qa-with-the-next-branch). It would be awesome to be able to use the same api as styled-components and have this utility methods playing nice together.

**Comment by jxnblk** [comment]
Unless `theme` is passed as a prop (which it looks like it's still only in context here https://github.com/cssinjs/styled-jss/pull/35/files#diff-04c6e90faac2675aa89e2176d2eec7d8R66), styled-system won't be able to pick up the theme values. That said, I don't see why this `styled-jss` library shouldn't work like every other styled-components clone

**Comment by hendrul** [comment]
Yes with the PR they are passing the theme, look at the end of the PR, 
```patch
+const Button = styled('button')((props, {theme}) => ({
+  color: theme.color,
+  'background-color': theme.backgroundColor,
```

**Comment by jxnblk** [comment]
Right, what I’m saying is that the convention is to pass context values as props rather than leaving them on the context object

**Comment by jxnblk** [comment]
Closing this out. Again, styled-system should work with any CSS-in-JS library that passes `props.theme` as a convention

**Comment by lttb** [comment]
@hendrul just to note, we've decided to move on with `props.theme` convention, so theming support had been released in v2.2.0

there is an example with `styled-system`: https://codesandbox.io/s/0p156o834n

@jxnblk btw awesome project, thanks!
